core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -327,6 +328,15 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
     }
     for (int i = rank_start; i != rank_end; i += increment) {
       const index_type length = m_upper[i] - m_lower[i];
+
+      if (m_upper[i] < m_lower[i]) {
+        std::stringstream msg;
+        msg << "Kokkos::MDRangePolicy bounds error: The lower bound ("
+            << m_lower[i] << ") is greater than its upper bound (" << m_upper[i]
+            << ") in rank " << i + 1 << ".";
+        Kokkos::abort(msg.str().c_str());
+      }
```

Comment:
Did you consider just treating it as an empty range if the upper bound is lower than the lower bound?
Isn't that what we are doing for `RangePolicy`?

Reply:
I didn't realize that `RangePolicy` behaves this way, but the idea of silently changing a policy to be an empty range seems unhelpful for the users, although I admit that given a lower bound that's higher than its upper bound, making it an empty range is most logical.

In case of `MDRangePolicy`, since the lower bound and upper bound checks will be needed for all ranks, should all ranges become empty ranges upon finding a failed bound check or should only failing ranks be empty ranges?

Reply:
I personally think we should do something that is the most useful for the users, which IMO is catching the problem as early as possible. Adopting the solution proposed in this PR seems good to me because the problem is caught immediately during the construction of the policy.

Reply:
FWIW, at first I was also thinking that aborting was the right thing to do; but just emptying the range is also a good solution, afterall when doing `for(int i=0; i<-10;i++)` this is valid code that may happen when implementation an algorithm where the bound are computed using analytical expressions, and the desired behavior is doing nothing.

Reply:
"when implementating an algorithm where the bound are computed using analytical expressions, and the desired behavior is doing nothing." what do you mean by: 
1. using analytical expressions,
2. the desired behavor is doing nothing

Reply:
I think we need to differentiate what concerns the algirithm vs how the MDRangePolicy works. 
I don't think it would be right to tie the implementation/semantics of the MDRangePolicy to something that pertains a certain algorithm. 
It should be up to the algorithm to translate its requirements into something that makes sense for MDRangePolicy. 
if you algorithm has meaning for a range like [15, -20], then it should be up to you to translate that into an empty MDRangePolicy. 
IMO, the MDRangePolicy should either accept a valid set of bounds or empty bounds but NOT "flipped" bounds. 
Does it make sense what I am trying to convey?

Reply:
I agree with your last option (3 bullets above) and that it should be clearly stated in the documentation.
IMHO, if one rank has upper bound < lower bound then the right behavior, is don't do any iteration at all; afterall this is what happen when you have two nest for loops, and one of them has invalid range.

Somehow, about your 3rd bullet above, when upper bound is < lower bound, it is not really an invalid range, just an empty range, but for a different reason.

The last point is that Range and MDRange policy should behave the same. If you think aborting is best in case of invalid range in MDRange, then Range policy should behave likewise. 

Finally, I don't have strong opinion here, just saying that first I was thinking aborting was the right behavior, but finally I can live fine with emptying the range.

Reply:
Maybe, we just discuss it on Wednesday in the meeting.